### ISSUE TITLE
Add support for pathlib.Path

### ISSUE THREAD

**Issue by jarekwg** [jschneier/django-storages#1019]
Django has made some moves in the past year to support `pathlib.Path()` with `FileSystemStorage` https://code.djangoproject.com/ticket/28428

For alignment, django-storages should head in the same direction.

Not proposing any breaking changes here (just yet), but would be nice if `Path` objects were acceptable parameters when feeding in keys and prefixes.

**Comment by tolomea** [comment]
For S3Boto3Storage we are currently wraping it with this
```
class BaseS3Storage(S3Boto3Storage):
    def _clean_name(self, name):
        name = str(name)  # convert any pathlib stuff to strings
        return super()._clean_name(name)
```
to deal with this

**Comment by jschneier** [comment]
This landed in #1200.
